static/examples/batch/batch-slicing-threads.scd

```diff
@@ -0,0 +1,86 @@
+// Refer to https://learn.flucoma.org/overviews/batch-processing
+(
+s.waitForBoot({
+    var drum = Buffer.read(s,File.realpath(FluidDataSet.class.filenameSymbol).dirname.withTrailingSlash ++ "../AudioFiles/Nicol-LoopE-M.wav");
+    var slices = Buffer(s); // The slicer (FluidBufOnsetSlice) will write into this buffer the samples at which slices are detected
+    var data = FluidDataSet(s); // A FluidDataSet() for storing the analysis of each slice (mean centroid)
+
+    // Create a condition that hangs while processing is still happening and unhangs when its complete.
+    var processingCompleted = Condition();
+    var counter = 0; // Also keep track of how many slices we have processed.
+    s.sync;
+
+    // Slice the drum Buffer() based on consecutive changes in spectral frames
+    // The samples at which slices are detected will be written into the "slices" Buffer()
+    FluidBufOnsetSlice.processBlocking(s, drum, indices:slices, metric:9,threshold:0.5);
```

Comment:
`s.sync` should go after this so that we know `slices` is actually populated before `loadToFloatArray`

Reply:
Not sure why we have `processBlocking` for this example (which is showcasing async approach) and `process` in the other (which is showcasing sync approach)

Reply:
Because I am bad at copy and paste :)

Do you think it is worth using a Semaphore here too? it's entirely possible to want to do a butt load of slices and maybe its good to show that you may need to cap the concurrent jobs on the server?

Reply:
I've put in a semaphore :)